compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -961,7 +963,7 @@ impl<'tcx> LayoutOfHelpers<'tcx> for CodegenCx<'_, 'tcx> {
     #[inline]
     fn handle_layout_err(&self, err: LayoutError<'tcx>, span: Span, ty: Ty<'tcx>) -> ! {
         if let LayoutError::SizeOverflow(_) = err {
-            self.sess().span_fatal(span, &err.to_string())
+            self.sess().emit_fatal(LayoutSizeOverflow { span, error: err.to_string() })
```

Comment:
`LayoutError` could implement `SessionDiagnostic` itself if we supported `SessionDiagnostic` on enums